manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/processor/CustomSnippetsProcessor.java

```diff
@@ -0,0 +1,111 @@
+ *
+ * @author Nishanth Panthangi
+ */
+
+@Component
+public class CustomSnippetsProcessor {
+    private static final Logger logger = LoggerFactory.getLogger(CustomSnippetsProcessor.class);
+
+    public Multimap<String,String> processCustomSnippetFiles(List<String> k8sSnippetFilePaths) throws HyscaleException {
+        if(k8sSnippetFilePaths == null){
+            return null;
+        }
+        Multimap<String,String> kindVsCustomSnippets = ArrayListMultimap.create(); // NOSONAR
+        String kind = null;
+        String yamlSnippet = null;
+        for(String snippetFilePath: k8sSnippetFilePaths){
+            File snippetFile = new File(snippetFilePath); //NOSONAR
+            try{
+                yamlSnippet = HyscaleFilesUtil.readFileData(snippetFile);
+                if(yamlSnippet == null){
+                    throw new HyscaleException(CommonErrorCode.FAILED_TO_READ_FILE);
+                }
+            }catch (HyscaleException e){
+                logger.error("Failed to read Custom Snippet file from path {}",snippetFilePath);
+                WorkflowLogger.error(ManifestGeneratorActivity.FAILED_TO_READ_CUSTOM_SNIPPETS,snippetFilePath);
+                throw new HyscaleException(CommonErrorCode.FAILED_TO_READ_FILE,snippetFilePath);
+            }
+            try {
+                // Validate YAML
+                String jsonSnippet =  DataFormatConverter.yamlToJson(yamlSnippet);
+                kind = identifyKind(jsonSnippet);
+            }catch(HyscaleException e){
+                logger.error("Failed to convert YAML Snippet to json, invalid YAML File {}",yamlSnippet,e);
+                WorkflowLogger.error(ManifestGeneratorActivity.INVALID_CUSTOM_SNIPPET,snippetFilePath);
+                throw new HyscaleException(CommonErrorCode.FAILED_TO_READ_FILE,snippetFilePath);
+            }
+            kindVsCustomSnippets.put(kind,yamlSnippet);
+        }
+        return kindVsCustomSnippets;
+    }
+
+    public String mergeYamls(String source, String patch) throws HyscaleException {
+        String mergedYaml = null;
+        MapFieldDataProvider mapFieldDataProvider = new MapFieldDataProvider(); //NOSONAR
+        source = DataFormatConverter.yamlToJson(source);
+        patch = DataFormatConverter.yamlToJson(patch);
+
+        String strategicMergeJson = StrategicPatch.apply(source, patch,mapFieldDataProvider);
+        try {
+            JsonNode jsonNode = ObjectMapperFactory.jsonMapper().readTree(strategicMergeJson); //NOSONAR
+            mergedYaml = ObjectMapperFactory.yamlMapper().writeValueAsString(jsonNode);
+            return mergedYaml;
+        }catch (IOException e){
+            logger.error("Error while converting merged json string to yaml ",e);
+        }
+        return mergedYaml;
+    }
+
+    public String identifyKind(String jsonSnippet){
+        ObjectMapper objectMapper = ObjectMapperFactory.yamlMapper(); //NOSONAR
+        JsonNode jsonNode = null;
+        String kind = null;
```

Comment:
Add null check for jsonSnippet
Similarly no need for kind variable
jsonNode can be defined in the try block